### DISCUSSION TITLE
HuggingFace Hub integrated in AllenNLP

### DISCUSSION THREAD

**Discussion by LysandreJik** [allenai/allennlp#5140] [Show and tell]
The PR https://github.com/allenai/allennlp/pull/5052 being merged, AllenNLP now supports loading files directly from the HuggingFace model hub!

By uploading the contents of an archive to the model hub, the files can now be browsed through, versioned, and the repository identifier can be used seamlessly within AllenNLP:

```diff
from allennlp.predictors.predictor import Predictor
import allennlp_models.rc

- checkpoint = "https://storage.googleapis.com/allennlp-public-models/bidaf-elmo.2021-02-11.tar.gz"
+ checkpoint = "hf://lysandre/bidaf-elmo-model-2020.03.19"
predictor = Predictor.from_path(checkpoint)
predictions = predictor.predict_json({
  "passage":
      "The Matrix is a 1999 science fiction action "
      "film written and directed by The Wachowskis, "
      "starring Keanu Reeves, Laurence Fishburne, "
      "Carrie-Anne Moss, Hugo Weaving, and Joe P"
      "antoliano.",
  "question":
      "Who stars in The Matrix?"
})
print(predictions["best_span_str"])
```

This also offers an inference API for the question answering models uploaded to the hub!
<div align="center">
<img src="https://user-images.githubusercontent.com/30755778/115630419-f65bad80-a2d1-11eb-8ef5-64756fe62c0f.png" alt="Inference API for BIDAF: A question answering model on the hub." width="400"/>
<div>

We look forward to hosting more models on the hub and enabling more use-cases through the inference API!

**Comment by julien-c** [comment]
- @allennlp team: let us know if we can help to add some documentation about this feature
- if you can push a few models to the hf.co hub and let us know if everything worked well, that would be awesome!
